frontend/src/components/AddConcepts.tsx

```diff
@@ -0,0 +1,134 @@
+  readonly distinct_person_cnt: string;
+}
+
+const columns = [
+  {
+    name: "Concept Name",
+    selector: (row) => row.concept_name,
+    sortable: true,
+    width: 200,
+  },
+  {
+    selector: row => row.domain_id,
+    name: "Domain",
+    sortable: true,
+    width: 200,
+  },
+  {
+    selector: row => row.total_cnt,
+    name: "Total Count",
+    sortable: true,
+    width: 100,
+  },
+  {
+    selector: row => row.distinct_person_cnt,
+    name: "Distinct Person Count",
+    sortable: true,
+    width: 100,
+  },
+];
+
+export function AddConcepts() {
+  return <ConceptStringSearch />;
+
+}
+// from https://stackoverflow.com/a/66167322/1368860
+function ConceptStringSearch() {
+  const dataGetter: DataGetter = useDataGetter();
+  let concepts: Concept[] = [];
+  const [searchText, setSearchText] = React.useState("");
+  const [concept_ids, setConceptIds] = React.useState(concepts);
+  const lastRequest = React.useRef(null);
+
+  // this effect will be fired every time searchText changes
+  React.useEffect(() => {
+    (async () => {
+      // setting min lenght for searchText
+      if (searchText.length >= 3) {
+        // updating the ref variable with the current searchText
+        lastRequest.current = searchText;
+        const r = await dataGetter.fetchAndCacheItems(dataGetter.apiCalls.concept_search, searchText)
+        if (lastRequest.current === searchText) {
+          console.log("response is valid!", r);
+          setConceptIds(r);
+        } else {
+          console.log("discarding api response", searchText, lastRequest.current);
+        }
+      }
+    })();
+  }, [searchText]);
+
+  return (
+    <div className="App" style={{paddingLeft: 100}}>
+      <h1>Concept Search</h1>
+      <input style={{width: 300}} type="text" placeholder="match characters in concept name"
+             onChange={(e) => setSearchText(e.target.value)}
+             value={searchText}
+             autoFocus={true}
+      />{'\u00A0'}{'\u00A0'}{'\u00A0'}{concept_ids.length ? concept_ids.length.toLocaleString() + ' concept_ids' : ""}
+      <ConceptTable concept_ids={concept_ids} />
+    </div>
+  );
+}
+
+function ConceptTable(props) {
+  let {concept_ids} = props;
+  const dataGetter = useDataGetter();
+  const [concepts, setConcepts] = useState([]);
+  const [loading, setLoading] = useState(false);
+  const totalRows = concept_ids.length;
+  const [perPage, setPerPage] = useState(10);
+
+  const fetchConcepts = async page => {
+    setLoading(true);
+    let ids = concept_ids.slice(page - 1, perPage);
+    let conceptLookup = await dataGetter.fetchAndCacheItems(dataGetter.apiCalls.concepts, ids);
+    const _concepts = ids.map(id => conceptLookup[id]);
+    setConcepts(_concepts);
+    setLoading(false);
+  };
+  const handlePageChange = page => {
+    fetchConcepts(page);
+  };
+  const handlePerRowsChange = async (newPerPage, page) => {
+    setLoading(true);
+    let ids = concept_ids.slice(page - 1, perPage);
```

Comment:
## Bug fix: incorrect, vanishing results on page change
```suggestion
    let ids = concept_ids.slice(page - 1, page - 1 + perPage);
```

Without the fix, I was seeing a situation where going to the next page didn't show any new concepts, but simply shortened the list of concepts already showing.

Page 1: 10 concepts
<img width="1055" alt="10" src="https://github.com/jhu-bids/TermHub/assets/13045020/bab1ff58-5717-416e-a74d-760994c32566">

Page 10: 1 concept (from the same original list)
<img width="796" alt="1" src="https://github.com/jhu-bids/TermHub/assets/13045020/d01badf6-d31c-4360-9097-a3594951ac5a">